charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -90,42 +94,72 @@ def configure_kafka(self, zk_units, log_dir=None):
         self.restart()
 
     def restart(self):
-        self.stop()
-        self.start()
+        '''
+        Restarts the Kafka service.
+        '''
+        host.service_restart(KAFKA_SERVICE)
 
     def start(self):
+        '''
+        Starts the Kafka service.
+        '''
         host.service_start(KAFKA_SERVICE)
 
     def stop(self):
+        '''
+        Stops the Kafka service.
+
+        '''
         host.service_stop(KAFKA_SERVICE)
-    
+
     def is_running(self):
+        '''
+        Restarts the Kafka service.
+        '''
         return host.service_running(KAFKA_SERVICE)
 
-def resolve_private_address(addr):
-    IP_pat = re.compile(r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}')
-    contains_IP_pat = re.compile(r'\d{1,3}[-.]\d{1,3}[-.]\d{1,3}[-.]\d{1,3}')
-    if IP_pat.match(addr):
-        return addr  # already IP
-    try:
-        ip = socket.gethostbyname(addr)
-        return ip
-    except socket.error as e:
-        hookenv.log(
-            'Unable to resolve private IP: %s (will attempt to guess)' %
-            addr,
-            hookenv.ERROR
-        )
-        hookenv.log('%s' % e, hookenv.ERROR)
-        contained = contains_IP_pat.search(addr)
-        if not contained:
-            raise ValueError('Unable to resolve or guess IP from private-address: %s' % addr)
-        return contained.groups(0).replace('-', '.')
+    def read_zk_peers(self):
```

Comment:
"Peers" has a specific meaning in Juju, referring to other units of the same application. I'd rename this to something else, like read_zk_nodes.